### PR TITLE
[PAN-2811] Be more lenient with discovery message deserialization

### PR DESCRIPTION
**Pull request by mbaxter**
## PR description
[EIP-8](https://eips.ethereum.org/EIPS/eip-8) specifies that devp2p discovery message deserialization should be lenient in order to make protocol upgrades easier.  This PR updates discovery packet deserialization to be more lenient by:
* ignoring extra fields
* ignoring the version in PING packets

### CODE REVIEW

File: ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/FindNeighborsPacketData.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public static FindNeighborsPacketData readFrom(final RLPInput in) {
     in.enterList();
     final BytesValue target = in.readBytesValue();
     final long expiration = in.readLongScalar();
-    in.leaveList();
+    in.leaveList(true);
```
Comment (RatanRSur) [inline ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/FindNeighborsPacketData.java]: (optional) make leaveList parameter an enum. Much easier to understand what
`in.leaveList(RLPInput.IgnoreRest)` does as opposed to `in.leaveList(true)`

Comment (shemnon) [reply ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/FindNeighborsPacketData.java]: I'd say either @RatanRSur's suggestion or at least leave a comment that (true) means to ignore the rest of the list, for all the `leaveList(true)` calls.

Comment (ajsutton) [reply ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/FindNeighborsPacketData.java]: Third option would be to have two methods - the usual `leaveList` and the new `leaveListSkippingRemainingElements`

Comment (mbaxter) [reply ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/FindNeighborsPacketData.java]: Changed this to `leaveListLenient()`

**Pull request merged**